File: src/test/java/org/dicio/skill/SkillComponentTest.java
package org.dicio.skill;

import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertSame;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.junit.Test;

public class SkillComponentTest {
    @Test
    public void testConstructorAndSetGet() {
        final SkillInfo skillInfo = new SkillInfo("id", 0, 0, 0, false) {
            @Override public boolean isAvailable(final SkillContext context) { return false; }
            @Override public Skill build(final SkillContext context) { return null; }
            @Nullable
            @Override public Fragment getPreferenceFragment() { return null; }
        };

        final SkillComponent skill = new SkillComponent() {};

        assertNull(skill.getSkillInfo());
        skill.setSkillInfo(skillInfo);
        assertSame(skillInfo, skill.getSkillInfo());

        final SkillContext skillContext = new SkillContext();
        skill.setContext(skillContext);
        assertSame(skillContext, skill.ctx());
    }
}

File: src/test/java/org/dicio/skill/SkillComponentTest.kt
package org.dicio.skill

import android.content.Context
import android.content.SharedPreferences
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.types.shouldBeSameInstanceAs
import org.dicio.numbers.ParserFormatter
import org.dicio.skill.output.SpeechOutputDevice
import java.util.Locale

class SkillComponentTest : StringSpec({
    "constructor, getters and setters" {
        val skill: SkillComponent = object : SkillComponent() {}

        skill.setContext(TestSkillContext)
        skill.ctx() shouldBeSameInstanceAs TestSkillContext
    }
})

object TestSkillContext : SkillContext {
    override val android: Context get() = TODO()
    override val preferences: SharedPreferences get() = TODO()
    override val locale: Locale get() = TODO()
    override val parserFormatter: ParserFormatter get() = TODO()
    override val speechOutputDevice: SpeechOutputDevice get() = TODO()
}
